src/assets/TokensController.test.ts

```diff
@@ -312,4 +293,458 @@ describe('TokensController', () => {
+          Promise.resolve({ supportsInterface: supportsInterfaceStub }),
+        );
+      const tokenAddress = '0xda5584cc586d07c7141aa427224a4bd58e64af7d';
+      tokensController.update({
+        tokens: [
+          {
+            address: tokenAddress,
+            symbol: 'TESTNFT',
+            decimals: 0,
+          },
+        ],
+      });
+
+      const result = await tokensController.updateTokenType(tokenAddress);
+
+      expect(result.isERC721).toBe(true);
+    });
+
+    it('should return true when token is in our contract-metadata repo', async function () {
+      const supportsInterfaceStub = sinon.stub().returns(Promise.resolve(true));
+      await sinon
+        .stub(tokensController, '_createEthersContract')
+        .callsFake(() =>
+          Promise.resolve({ supportsInterface: supportsInterfaceStub }),
+        );
+      const tokenAddress = '0x06012c8cf97BEaD5deAe237070F9587f8E7A266d';
+
+      const result = await tokensController._detectIsERC721(tokenAddress);
+      expect(result).toBe(true);
+    });
+
+    it('should return true when the token is not in our contract-metadata repo but tokenContract.supportsInterface returns true', async function () {
+      const supportsInterfaceStub = sinon.stub().returns(Promise.resolve(true));
+      await sinon
+        .stub(tokensController, '_createEthersContract')
+        .callsFake(() =>
+          Promise.resolve({ supportsInterface: supportsInterfaceStub }),
+        );
+
+      const tokenAddress = '0xda5584cc586d07c7141aa427224a4bd58e64af7d';
+
+      const result = await tokensController._detectIsERC721(tokenAddress);
+
+      expect(result).toBe(true);
+    });
+
+    it('should return false when the token is not in our contract-metadata repo and tokenContract.supportsInterface returns false', async function () {
+      const supportsInterfaceStub = sinon
+        .stub()
+        .returns(Promise.resolve(false));
+      await sinon
+        .stub(tokensController, '_createEthersContract')
+        .callsFake(() =>
+          Promise.resolve({ supportsInterface: supportsInterfaceStub }),
+        );
+      const tokenAddress = '0xda5584cc586d07c7141aa427224a4bd58e64af7d';
+
+      const result = await tokensController._detectIsERC721(tokenAddress);
+      expect(result).toBe(false);
+    });
+  });
+
+  describe('on watchAsset', function () {
+    let asset: any, type: any;
+
+    beforeEach(function () {
+      type = 'ERC20';
+      asset = {
+        address: '0x000000000000000000000000000000000000dEaD',
+        decimals: 12,
+        symbol: 'SES',
+        image: 'image',
+      };
+      const supportsInterfaceStub = sinon
+        .stub()
+        .returns(Promise.resolve(false));
+      sinon
+        .stub(tokensController, '_createEthersContract')
+        .callsFake(() =>
+          Promise.resolve({ supportsInterface: supportsInterfaceStub }),
+        );
+    });
+
+    afterEach(function () {
+      sinon.restore();
+    });
+
```

Comment:
This seems redundant, since `sinon.restore()` is called in the `beforeEach` in the top-level `describe` block

```suggestion
```